src/Client.ts

```diff
@@ -272,6 +271,50 @@ export default class Client {
     return newBundle
   }
 
+  /**
+   * Identical to getUserContact but for multiple peer addresses
+   */
+  async getUserContacts(
+    peerAddresses: string[]
+  ): Promise<(PublicKeyBundle | SignedPublicKeyBundle | undefined)[]> {
+    // EIP55 normalize all peer addresses
+    const normalizedAddresses = peerAddresses.map((address) =>
+      utils.getAddress(address)
+    )
+    // The logic here is tricky because we need to do a batch query for any uncached bundles,
+    // then interleave back into an ordered array. So we create a map<string, keybundle|undefined>
+    // and fill it with cached values, then take any undefined entries and form a BatchQuery from there.
+    const addressToBundle = new Map<
+      string,
+      PublicKeyBundle | SignedPublicKeyBundle | undefined
+    >()
+    const uncachedAddresses = []
+    for (const address of normalizedAddresses) {
+      const existingBundle = this.knownPublicKeyBundles.get(address)
```

Comment:
I see us reading from `knownPublicKeyBundles`, but don't see us writing the results back in. WDYT about storing the results to save on future requests.

Reply:
Good call, I'll make the change